src/module_wrap.cc

```diff
@@ -410,8 +410,7 @@ void ModuleWrap::Evaluate(const FunctionCallbackInfo<Value>& args) {
 
   // Convert the termination exception into a regular exception.
   if (timed_out || received_signal) {
-    if (!env->is_main_thread() && env->is_stopping())
-      return;
+    if (!env->is_main_thread() && !env->can_call_into_js()) return;
```

Comment:
(Similar comment here, this is to ensure that the termination exception as part of stopping the environment passes through)